File: CS/WinForms_MapControl_KmlFileDataAdapter/Form1.cs
using DevExpress.XtraMap;
using System;
using System.Windows.Forms;

namespace WinForms_MapControl_KmlFileDataAdapter {
    public partial class Form1 : Form {
        const string filePath = "../../kmlFile.kml";

        VectorItemsLayer KmlLayer { get { return (VectorItemsLayer)mapControl1.Layers["KmlLayer"]; } }

        public Form1() {
            InitializeComponent();

            #region #KmlFileDataAdapter
            // Create a KML file data adapter.
            Uri baseUri = new Uri(System.Reflection.Assembly.GetEntryAssembly().Location);
            KmlLayer.Data =  new KmlFileDataAdapter() {
                FileUri = new Uri(baseUri, filePath)
            };
            #endregion #KmlFileDataAdapter
        }
    }
}

File: VB/WinForms_MapControl_KmlFileDataAdapter/Form1.vb
Imports DevExpress.XtraMap
Imports System
Imports System.Windows.Forms

Namespace WinForms_MapControl_KmlFileDataAdapter

    Public Partial Class Form1
        Inherits Form

        Const filePath As String = "../../kmlFile.kml"

        Private ReadOnly Property KmlLayer As VectorItemsLayer
            Get
                Return CType(mapControl1.Layers("KmlLayer"), VectorItemsLayer)
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
'#Region "#KmlFileDataAdapter"
            ' Create a KML file data adapter.
            Dim baseUri As Uri = New Uri(Reflection.Assembly.GetEntryAssembly().Location)
            KmlLayer.Data = New KmlFileDataAdapter() With {.FileUri = New Uri(baseUri, filePath)}
'#End Region  ' #KmlFileDataAdapter
        End Sub
    End Class
End Namespace
